extension/js/background.js

```diff
@@ -42,58 +51,88 @@ function acceptInput(text, disposition) {
-    case "newForegroundTab":
-        chrome.tabs.create({url: text});
-        break;
-    case "newBackgroundTab":
-        chrome.tabs.create({url: text, active: false});
-        break;
+        case "currentTab":
+            chrome.tabs.update({
+                url: text
+            });
+            break;
+        case "newForegroundTab":
+            chrome.tabs.create({
+                url: text
+            });
+            break;
+        case "newBackgroundTab":
+            chrome.tabs.create({
+                url: text,
+                active: false
+            });
+            break;
     }
 }
 
 function init() {
 
-    /*db = new PouchDB('main')
-    //db.plugin()
-    var remoteCouch = false;*/
-
+    db = new PouchDB('main')
+        // db.plugin()
+    var remoteCouch = false;
+    db.search({
+        fields: ['text'],
+        build: true
+    }).then(function (info) {
+        // if build was successful, info is {"ok": true}
+    }).catch(function (err) {
+        // handle error
+        console.log(err);
+    });
     window.preloaded = [];
     window.cache = {};
-    chrome.storage.local.get(['blacklist', 'preferences'], function(items) {
+    chrome.storage.local.get(['blacklist', 'preferences'], function (items) {
         var obj = items['blacklist'];
         if (obj === undefined || !('PAGE' in obj && 'SITE' in obj && 'REGEX' in obj)) {
-            window.blacklist = {'PAGE':[], 'REGEX':[], 'SITE':[]}; // show example in page
-            chrome.storage.local.set({'blacklist':blacklist});
+            window.blacklist = {
+                'PAGE': [],
+                'REGEX': [],
+                'SITE': []
+            }; // show example in page
+            chrome.storage.local.set({
+                'blacklist': blacklist
+            });
         } else {
             window.blacklist = obj;
         }
 
         var obj = items['preferences'];
         if (obj === undefined) {
             window.preferences = {};
-            chrome.storage.local.set({'preferences':preferences});
+            chrome.storage.local.set({
+                'preferences': preferences
+            });
         } else {
             window.preferences = obj;
         }
     });
 
-    chrome.storage.local.get('index', function(items) {
+    chrome.storage.local.get('index', function (items) {
         var obj = items['index'];
         if (obj === undefined) {
             window.timeIndex = [];
-            chrome.storage.local.get(null, function(items) {
+            chrome.storage.local.get(null, function (items) {
                 for (var key in items) {
                     if (key != 'index') {
+                        // if(items[key].time != null)
                         timeIndex.push(items[key].time.toString());
                     }
                 }
 
-                timeIndex.sort(function(a,b) {return parseInt(a) - parseInt(b)}); // soonest last
+                timeIndex.sort(function (a, b) {
```

Comment:
As far as I know, this is a relict from the old search implementation, This function probably can be removed entirely.